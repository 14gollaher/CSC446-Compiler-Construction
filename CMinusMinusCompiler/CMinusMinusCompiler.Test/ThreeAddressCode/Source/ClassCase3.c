int test1()
{
	int a, b, c;
	a = 5;
	b = 10;
	c = a + b;
	return -1;
}

int main()
{
	int x;
	x = test1();
	return -1;
}

/*NOTE: blindly copied paste my output to this results, so if wrong check.*/