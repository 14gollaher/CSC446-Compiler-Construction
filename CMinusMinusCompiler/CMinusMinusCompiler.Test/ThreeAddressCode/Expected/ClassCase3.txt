Proc test1
	_bp-8 = 5
	_bp-2 = _bp-8
	_bp-10 = 10
	_bp-4 = _bp-10
	_bp-12 = _bp-2 + _bp-4
	_bp-6 = _bp-12
	_bp-14 = -1
	_AX = _bp-14
Endp test1

Proc main
	Call test1
	_bp-2 = _AX
	_bp-4 = -1
	_AX = _bp-4
Endp main

Start Proc Main
